LoopKitUI/View Controllers/GlucoseRangeScheduleTableViewController.swift

```diff
@@ -168,147 +350,189 @@ public class GlucoseRangeScheduleTableViewController: DailyValueScheduleTableVie
+
+        return indexPath
+    }
+
+    public override func tableView(_ tableView: UITableView, didDeselectRowAt indexPath: IndexPath) {
+        tableView.beginUpdates()
+        tableView.endUpdates()
     }
 
     public override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        switch Section(rawValue: indexPath.section)! {
-        case .schedule:
-            super.tableView(tableView, didSelectRowAt: indexPath)
-        case .override:
+        switch sections[indexPath.section] {
+        case .save:
+            delegate?.saveSchedule(for: self, completion: { (result) in
+                switch result {
+                case .success:
+                    self.isScheduleModified = false
+                    self.updateInsertButton()
+                case .failure(let error):
+                    self.present(UIAlertController(with: error), animated: true)
+                }
+            })
+        default:
             break
         }
+        tableView.endEditing(false)
+        tableView.endUpdates()
+        tableView.deselectRow(at: indexPath, animated: true)
     }
 
-    // MARK: - RepeatingScheduleValueTableViewCellDelegate
+    public override func tableView(_ tableView: UITableView, targetIndexPathForMoveFromRowAt sourceIndexPath: IndexPath, toProposedIndexPath proposedDestinationIndexPath: IndexPath) -> IndexPath {
 
-    override public func datePickerTableViewCellDidUpdateDate(_ cell: DatePickerTableViewCell) {
-        if let indexPath = tableView.indexPath(for: cell) {
-            let currentItem = scheduleItems[indexPath.row]
+        guard sourceIndexPath.section == proposedDestinationIndexPath.section else {
+            return sourceIndexPath
+        }
 
-            scheduleItems[indexPath.row] = RepeatingScheduleValue(
-                startTime: cell.date.timeIntervalSince(midnight),
-                value: currentItem.value
-            )
+        guard sourceIndexPath != proposedDestinationIndexPath else {
+            return proposedDestinationIndexPath
         }
 
-        super.datePickerTableViewCellDidUpdateDate(cell)
-    }
+        let indices = insertableIndices(removing: sourceIndexPath.row)
 
-    func repeatingScheduleValueTableViewCellDidUpdateValue(_ cell: RepeatingScheduleValueTableViewCell) {
-        if let indexPath = tableView.indexPath(for: cell), let cell = cell as? GlucoseRangeTableViewCell {
-            let currentItem = scheduleItems[indexPath.row]
+        if indices[proposedDestinationIndexPath.row] {
+            return proposedDestinationIndexPath
+        } else {
+            var closestRow = sourceIndexPath.row
 
-            scheduleItems[indexPath.row] = RepeatingScheduleValue(startTime: currentItem.startTime, value: DoubleRange(minValue: cell.minValue, maxValue: cell.value))
+            for (index, valid) in indices.enumerated() where valid {
+                if abs(proposedDestinationIndexPath.row - index) < closestRow {
```

Comment:
I've stared at this for 5 minutes and can't get my head around the logic. It feels like something is wrong here, comparing a relative value (`abs(proposedDestinationIndexPath.row - index)`) against an absolute value (`closestRow`), but I can't put my finger on it. I keep trying to figure it out if the rows/indexes are all very large and see how the math works.

Reply:
Yep, you're right. It should be compared against `abs(proposedDestinationIndexPath.row - closestRow)`, so we're comparing apples to apples. Checked in an extension that can be reused by the various places that need to do this calc.